src/System.IO.FileSystem/src/System/IO/Win32FileStreamAsyncResult.cs

```diff
@@ -1,92 +1,49 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using System;
-using Microsoft.Win32;
 using Microsoft.Win32.SafeHandles;
 using System.Security;
 using System.Threading;
 using System.Threading.Tasks;
 using System.Runtime.InteropServices;
-using System.Runtime.CompilerServices;
-using System.Globalization;
-using System.Runtime.Versioning;
 using System.Diagnostics;
 
 namespace System.IO
 {
     internal partial class Win32FileStream
     {
+#if USE_OVERLAPPED
         // This is an internal object implementing IAsyncResult with fields
         // for all of the relevant data necessary to complete the IO operation.
         // This is used by AsyncFSCallback and all of the async methods.
         unsafe private sealed class FileStreamAsyncResult : IAsyncResult
```

Comment:
The problem is that the Overlapped API only allows you to stow away an IAsyncResult (in the [AsyncResult property](https://msdn.microsoft.com/en-us/library/system.threading.overlapped.asyncresult(v=vs.110).aspx)) - so I have no other way to pass the object to the callback unless it implements the interface.
(Although, admittedly, I could just throw for all of the interface calls)